core-addon/Core.js

```diff
@@ -493,6 +497,23 @@ module.exports = class Core {
       await browser.management.getAll().then(addons =>
         addons.filter(a => a.type == "extension")
           .map(a => a.id));
+
+    // Attempt to resume any paused studies, or pause any running
+    // studies, as appropriate.
+    try {
+      for (const study of studies) {
+        if (installedAddonsIds.includes(study.addonId)) {
+          if (study.studyPaused) {
+            await this._sendMessageToStudy(study.addonId, "pause", {});
+          } else {
+            await this._sendMessageToStudy(study.addonId, "resume", {});
+          }
+        }
+      }
+    } catch (err) {
+      console.error("Changing study state failed:", err);
+    }
+
```

Comment:
Can we break this to a separate function?

Reply:
I was thinking I could probably shrink this down if I use a more functional style too :) I'll see what I can come up with.